src/Features/Lsif/Generator/Generator.cs

```diff
@@ -74,13 +81,14 @@ public void GenerateForCompilation(Compilation compilation, string projectPath,
         /// lets us link symbols across files, and will only talk about "top level" symbols that aren't things like locals that can't
         /// leak outside a file.
         /// </remarks>
-        private static Id<Graph.LsifDocument> GenerateForDocument(
-            SemanticModel semanticModel,
+        private static async Task<Id<Graph.LsifDocument>> GenerateForDocumentAsync(
```

Comment:
(note other comments about intentional lack of async and workspace types here...)